loader/src/main/java/eu/europeana/fulltext/loader/service/XMLParserService.java

```diff
@@ -256,7 +264,7 @@ private boolean parseAnnotation(XMLEventReader reader, StartElement annotationEl
                     progressAnnotation.addItemFail();
                 }
             }
-        } catch (LoaderException e) {
+        } catch (LoaderException | ParseException e) {
```

Comment:
The idea behind the 'LoaderException' class is that we only throw that so other classes who use this don't have to catch all kinds of exceptions (plus SonarQube warns about this I think). So I propose to catch where ever it is the ParseException is thrown and rethrow it as a LoaderException.

Reply:
Ah, indeed. This is now gone, after I rewrote the media part of the createTarget() method.

Reply:
SonarLint complained about some other issues (no errors, just warnings), among which one that I know I sometimes am guilty of: the superfluous assignment of this kind: 

Gizmo g = new Gizmo(initialData)
g = calibrateGizmo(g);

private Gizmo calibrateGizmo(Gizmo gizmo){
   gizmo.addBigRedButton("HERE");
   gizmo.lubricateWell(new Oil(expensive));
   gizmo.attachLabel("Don't press this button");
   return gizmo;
}

.. or any other thing returned but never used. This happened in this class in line 273 - returning the boolean "result" that was never used.

Also, there was a dead "else" branch. 

I removed both.